.github/workflows/e2e.yaml

```diff
@@ -30,11 +30,11 @@ jobs:
         run: |
           set -euo pipefail
           go clean -testcache
+          make gofail-enable
 
           echo "${TARGET}"
           case "${TARGET}" in
             linux-amd64-e2e)
-              make gofail-enable
```

Comment:
This change causes a big change in convention. I would recommend to not change this, but add a skip in tests if etcd binary was not complied with gofail enabled.

E2e tests should be runnable on any etcd binary, allowing us to do run them against old releases or the exact binaries we are releasing. This creates a problem, some tests like yours require gofail a custom build binaries. How do we just expect some e2e tests to fail now?

The approach I have previously proposed was to skip the test that cannot be executed for some reason. That way you can still run `make test-e2e` and get accurate results and CI can just compile in `gofail` to run all tests. I would suggest taking this approach.

The one remaining question is "If we are skipping some tests, how do we know we don't forget any tests?". If someone removes or breaks `make gofail-enable` then we are screwed. To mitigate this risk I think we should consider adding validation in CI that ensures that etcd binary tested has expected failpoints.

Reply:
> This change causes a big change in convention

Just to clarify the change you meant is moving `make gofail-enable` from line 37 to line 33 that causing both `amd64` and `386` will run e2e tests with gofail enabled and compiled etcd binary, correct? 

> E2e tests should be runnable on any etcd binary, allowing us to do run them against old releases or the exact binaries we are releasing. This creates a problem, some tests like yours require gofail a custom build binaries. How do we just expect some e2e tests to fail now?

> The approach I have previously proposed was to skip the test that cannot be executed for some reason. That way you can still run make test-e2e and get accurate results and CI can just compile in gofail to run all tests. I would suggest taking this approach.

Thanks for the detailed explanation!!

I did not think it through when making this change and hope the CI would pass because local run did not fail to set up the failpoint. 

I am thinking for each test, it could opt-in to create etcd process with gofail enabled and compiled binary. If the binary does not exist, then skip. By default e2e test will use the vanilla binary even in the CI. This should resolve the problem "test the exact binaries we are releasing". 

> To mitigate this risk I think we should consider adding validation in CI that ensures that etcd binary tested has expected failpoints.

Just some naive thought, if the test itself can tell it is running on github action or k8s CI framework with some environment variables, then it should fail to guarantee we don't forget any tests.